Commands out of sync; you can't run this command now

simply ran following code on my ubuntu-20 machine running with mysql server bug got below error
(python version 3.8.2)
-----------------------------------------------------
def test_using_postgresql(mysql_proc, mysql):
    mysql.query("SELECT version();")
-----------------------------------------------------

(common-venv-3.8.2) /home/ravi/cprog/for-mysql>pytest 
======================================================================================= test session starts ========================================================================================
platform linux -- Python 3.8.2, pytest-5.4.2, py-1.8.1, pluggy-0.13.1
rootdir: /home/ravi/cprog/for-mysql
plugins: mysql-2.0.1, flask-1.0.0, postgresql-2.4.0
collected 1 item                                                                                                                                                                                   

test_w.py .E                                                                                                                                                                                 [100%]

============================================================================================== ERRORS ==============================================================================================
____________________________________________________________________________ ERROR at teardown of test_using_postgresql ____________________________________________________________________________

    def drop_database():
>       mysql_conn.query('DROP DATABASE IF EXISTS %s' % mysql_db)

../../.pyenv/versions/3.8.2/envs/common-venv-3.8.2/lib/python3.8/site-packages/pytest_mysql/factories/client.py:95: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <_mysql.connection open to 'localhost' at 0x55df362acf20>, query = 'DROP DATABASE IF EXISTS test'

    def query(self, query):
        # Since _mysql releases GIL while querying, we need immutable buffer.
        if isinstance(query, bytearray):
            query = bytes(query)
>       _mysql.connection.query(self, query)
E       MySQLdb._exceptions.ProgrammingError: (2014, "Commands out of sync; you can't run this command now")

../../.pyenv/versions/3.8.2/envs/common-venv-3.8.2/lib/python3.8/site-packages/MySQLdb/connections.py:259: ProgrammingError
===================================================================================== short test summary info ======================================================================================
ERROR test_w.py::test_using_postgresql - MySQLdb._exceptions.ProgrammingError: (2014, "Commands out of sync; you can't run this command now")
=================================================================================== 1 passed, 1 error in 17.84s ===================================================================

--- a/.gitignore
+++ b/.gitignore
@@ -18,6 +18,7 @@ develop-eggs
 lib
 lib64
 .cache/*
+venv/
 
 # Installer logs
 pip-log.txt

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -4,7 +4,9 @@ CHANGELOG
 unreleased
 -------
 
-- [enhnacement] Require at least mirakuru 2.3.0 - forced by changed stop method paramters change
+- [enhancement] Preemptively read data after each test in mysql client fixture.
+  This will make test run if the test itself forgot to fetch queried data.
+- [enhnacement] Require at least mirakuru 2.3.0 - forced by changed stop method parameters change
 
 2.0.1
 -------

--- a/src/pytest_mysql/factories/client.py
+++ b/src/pytest_mysql/factories/client.py
@@ -90,12 +90,11 @@ def mysql(process_fixture_name, passwd=None, dbname=None,
         )
         mysql_conn.query('USE %s' % mysql_db)
 
-        def drop_database():
-            mysql_conn.query('DROP DATABASE IF EXISTS %s' % mysql_db)
-            mysql_conn.close()
+        yield mysql_conn
 
-        request.addfinalizer(drop_database)
-
-        return mysql_conn
+        # clean up after test that forgot to fetch selected data
+        mysql_conn.store_result()
+        mysql_conn.query('DROP DATABASE IF EXISTS %s' % mysql_db)
+        mysql_conn.close()
 
     return mysql_fixture

--- a/tests/test_mysql.py
+++ b/tests/test_mysql.py
@@ -18,6 +18,11 @@ def test_mysql(mysql):
     cursor.close()
 
 
+def test_mysql_test_without_cursor(mysql):
+    """Run test without cursor and without fetching the data."""
+    mysql.query("SELECT VERSION();")
+
+
 def test_mysql_newfixture(mysql, mysql2):
     """More complext test with several mysql_processes."""
     cursor = mysql.cursor()

